src/tools/miri/tests/pass-dep/libc/libc-time.rs

```diff
@@ -315,3 +329,101 @@ fn test_localtime_r_multiple_calls_deduplication() {
         NUM_CALLS - 1
     );
 }
+
+fn test_nanosleep() {
+    let start_test_sleep = Instant::now();
+    let duration_zero = libc::timespec { tv_sec: 0, tv_nsec: 0 };
+    let remainder = ptr::null_mut::<libc::timespec>();
+    let is_error = unsafe { libc::nanosleep(&duration_zero, remainder) };
+    assert_eq!(is_error, 0);
+    assert!(start_test_sleep.elapsed() < Duration::from_millis(10));
+
+    let start_test_sleep = Instant::now();
+    let duration_100_millis = libc::timespec { tv_sec: 0, tv_nsec: 1_000_000_000 / 10 };
+    let remainder = ptr::null_mut::<libc::timespec>();
+    let is_error = unsafe { libc::nanosleep(&duration_100_millis, remainder) };
+    assert_eq!(is_error, 0);
+    assert!(start_test_sleep.elapsed() > Duration::from_millis(100));
+}
+
+#[cfg(any(
+    target_os = "freebsd",
+    target_os = "linux",
+    target_os = "android",
+    target_os = "solaris",
+    target_os = "illumos"
+))]
+mod test_clock_nanosleep {
+    use super::*;
+
+    /// Helper function used to create an instant in the future
+    fn add_100_millis(mut ts: libc::timespec) -> libc::timespec {
+        const SECOND: i64 = 1_000_000_000;
+        ts.tv_nsec += SECOND / 10;
+        ts.tv_sec = ts.tv_nsec / SECOND;
```

Comment:
Isn't this still wrong?
```suggestion
        ts.tv_sec += ts.tv_nsec / SECOND;
```

Reply:
It should be this now: 
```
fn add_100_millis(mut ts: libc::timespec) -> libc::timespec {
    const SECOND: i64 = 1_000_000_000;
    ts.tv_nsec += SECOND / 10;
    ts.tv_sec = ts.tv_nsec / SECOND;
    ts.tv_nsec %= SECOND;
    ts
}
```

Reply:
But that's wrong. For the reasons I already explained above.

Reply:
the overflow of `tv_nsec` is taken care of by:
```rust
ts.tv_nsec %= SECOND;
```

right?
I tested it in the playground and there it seems to work https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=8c8dcb8b0f9c7238b747b61e073ddc14

Reply:
Better take a bit more time and think things through twice next time. :)

My suggestion above has a comment, I think that's still a useful comment to explain the (clearly non-obvious!) logic. Please add that comment.